commonItems/Parser.cs

```diff
@@ -173,32 +219,54 @@ private class RegisteredRegex : RegisteredKeywordOrRegex {
 			return sb.ToString();
 		}
 
-		public static string? GetNextTokenWithoutMatching(BufferedReader reader) {
-			return reader.EndOfStream ? null : GetNextLexeme(reader);
+		public string? GetNextTokenWithoutMatching(BufferedReader reader) {
```

Comment:
Why are you matching in GetNextTokenWithoutMatching?

Reply:
WithoutMatching refers to matching against registered rules, here I was only matching against variable and interpolated expression regexes to resolve them before returning. Will move that to GetNextLexeme, though.

Reply:
Added a comment to the function.